### ISSUE TITLE
Inquiry Regarding the Optimization of Uncertainty Field in DebSDF

### ISSUE THREAD

**Issue by alvinjinsung** [DavidXu-JJ/DebSDF#9]
I am working on SDF-based volume rendering for precise surface reconstruction and recently came across your impressive work, DebSDF. 
I found it highly insightful and believe it could provide valuable direction for my research.

That said, I have a question regarding the optimization of the uncertainty field in DebSDF, which has been troubling me. 
From my understanding, the paper and code indicate that during the initial 10k iterations, the uncertainty field is not utilized. 
Even after this initial phase, the uncertainty computed by the network is directly used for masking, depth and normal loss, as well as ray sampling.

What I find unclear is how this uncertainty field—indicating the variance corresponding to the monocular prior—is being optimized in a way that aligns with our expectations. 
Since the only loss for uncertainty is the log term for uncertainty scores, I do not understand how this uncertainty field is being optimized as we expect.

I would greatly appreciate it if you could clarify how the training process ensures the optimization of the uncertainty field for reliable results.

Thank you!

**Comment by DavidXu-JJ** [comment]
Hi Alvin,

Thanks for your attention to our work.

About the optimization you mention,

For the first 10,000 iterations, the uncertainty field is not set to learn because the $$\hat D$$ and $$\hat N$$ in Equ.10 and 11 are still not stable at the initial stage. Learning a correct SDF field and a correct uncertainty field at the same time is not so much practicable.

<img width="456" alt="image" src="https://github.com/user-attachments/assets/8a1d4019-e7ef-420c-8f25-b97755a759f0" />


After 10,000 iterations, the uncertainty field will start learning. From the blue words in page 10:

<img width="1350" alt="image" src="https://github.com/user-attachments/assets/014e731a-bf6d-4059-b38c-51cfa3069fe9" />

The first 40,000 iterations don't use the uncertainty field for ray sampling and smoothing. (Because when the uncertainty field is still learning, it is not reliable to be used for better ray sampling and smoothing) However, the model is learning uncertainty fields between 10,000 and 40,000 iterations.

So I personally think

> Even after this initial phase, the uncertainty computed by the network is directly used for masking, depth and normal loss, as well as ray sampling.

you mention is a small misunderstanding here.

Hope my answer can solve your questions.

Thanks again for your attention to our work.

Best,
Jingwei

**Comment by alvinjinsung** [comment]
Hi Jingwei,
Thank you for your answer!

I understand that uncertainty field will be learned between 10,000 and 40,000 iteration and onward.

The more critical question for me is how does this uncertainty field learn correctly?
For it to work properly, uncertainty field should model the variance corresponding to the monocular prior appropriately.
What I do not understand is how does loss stated in equation 10 and 11 sufficient enough to learn correct uncertainty field?
The loss function from uncertainty score perspective only have the log term and denominator for scaling and I do not clearly understand how this loss can optimize uncertainty field as we expect.

I think I am not fully understanding the insight here and I wish you could share yours.

Thank you!

Best,
Alvin

**Comment by DavidXu-JJ** [comment]
Hi Alvin,

The modeling of this loss term is basically Gaussian negative likelihood. (The monocular geometry prediction is considered to be uncertain here.) The learning of the uncertainty field is driven by the inconsistent monocular prediction from multi-view observation.

This post shares a more detailed description and discussion of the formulation of it. (And the derivations like in this post are usually treated as preliminary, in my understanding.) You may find it helpful:

https://stats.stackexchange.com/questions/521091/optimizing-gaussian-negative-log-likelihood

More materials you may find helpful:

https://www.gatsby.ucl.ac.uk/~balaji/balaji-uncertainty-talk-cifar-dlrl.pdf

Best,
Jingwei

**Comment by alvinjinsung** [comment]
Thank you for valuable comments!
It really helped me understand it fully!

Thank you once again for great work!!

Best,
Alvin

**Comment by DavidXu-JJ** [comment]
Glad that helps!

Good luck with your work.

Issue closed.
